docs/directives/link.md

```diff
@@ -59,6 +59,107 @@ The `@link` directive requires specifying a source `src`, the resource's type `t
 
 - `headers`: This is an optional field that assigns custom headers to the gRPC reflection server requests. Specifying a key-value map of header names and their values achieves this. (Values supports Mustache template)
 
+### Linking other configs
+
+With `@link` you can link other [configuration files](#config) that will be merged with the root configuration i.e. the first config file you've specified for the cli.
+
+See in details how it works:
+
+- The definitions `@server`, `@upstream`, `@telemetry`, `@links` are used only from the root configuration and their definitions from the linked configs are ignored
+- The schema definitions (i.e. types, unions, enums) are merged by the [federation composition rules](https://www.apollographql.com/docs/graphos/reference/federation/composition-rules)
```

Comment:
If server,upstream,telemetry is removed from the .rc files we don't need to document them as directives.

Reply:
We need a separate section I think to document advanced server settings using the YAML format. I also feel we don't need to document JSON examples.

Reply:
I moved these directives to separate `config` folder and replaced it's content from directives to yaml configuration. Also created a root file for runtime configuration. 

There a still a lot of mentions in docs about these directives that we can update next to represent new look of the configurations